enterprise/server/remote_execution/containers/ociruntime/ociruntime.go

```diff
@@ -0,0 +1,595 @@
+				Options:     []string{"bind", "rprivate"},
+			},
+			{
+				Destination: "/sys/fs/cgroup",
+				Type:        "cgroup",
+				Source:      "cgroup",
+				Options:     []string{"rprivate", "nosuid", "noexec", "nodev", "relatime", "ro"},
+			},
+			{
+				Destination: execrootPath,
+				Type:        "bind",
+				Source:      c.workDir,
+				Options:     []string{"bind", "rprivate"},
+			},
+		},
+		Annotations: map[string]string{
+			// Annotate with podman's default stop signal.
+			// TODO: is this strictly needed?
+			"org.opencontainers.image.stopSignal": syscall.SIGTERM.String(),
+		},
+		Linux: &specs.Linux{
+			// TODO: set up cgroups
+			CgroupsPath: "",
+			Namespaces: []specs.LinuxNamespace{
+				{Type: specs.PIDNamespace},
+				{Type: specs.IPCNamespace},
+				{Type: specs.UTSNamespace},
+				{Type: specs.MountNamespace},
+				{Type: specs.CgroupNamespace},
+				// TODO: setup networking and set the correct namespace path
+				// here
+				{Type: specs.NetworkNamespace},
+			},
+			Seccomp: &seccomp,
+			Devices: []specs.LinuxDevice{},
+			Resources: &specs.LinuxResources{
+				// TODO: networking
+				Network: nil,
+			},
+			// TODO: grok MaskedPaths and ReadonlyPaths - just copied from podman.
+			MaskedPaths: []string{
+				"/proc/acpi",
+				"/proc/kcore",
+				"/proc/keys",
+				"/proc/latency_stats",
+				"/proc/timer_list",
+				"/proc/timer_stats",
+				"/proc/sched_debug",
+				"/proc/scsi",
+				"/sys/firmware",
+				"/sys/fs/selinux",
+				"/sys/dev/block",
+			},
+			ReadonlyPaths: []string{
+				"/proc/asound",
+				"/proc/bus",
+				"/proc/fs",
+				"/proc/irq",
+				"/proc/sys",
+				"/proc/sysrq-trigger",
+			},
+		},
+	}
+
+	// Provision devices based on host device node info.
+	// TODO: set up devices the same way podman does.
+	for _, path := range []string{
+		"/dev/null",
+		"/dev/zero",
+		"/dev/random",
+		"/dev/urandom",
+	} {
+		d, err := statDevice(path)
+		if err != nil {
+			return nil, fmt.Errorf("get %s device spec: %w", path, err)
+		}
+		spec.Linux.Devices = append(spec.Linux.Devices, *d)
+		spec.Linux.Resources.Devices = append(spec.Linux.Resources.Devices, specs.LinuxDeviceCgroup{
+			Allow:  true,
+			Access: "rw",
+			Type:   d.Type,
+			Major:  &d.Major,
+			Minor:  &d.Minor,
+		})
+	}
+
+	return &spec, nil
+}
+
+func (c *ociContainer) invokeRuntimeSimple(ctx context.Context, args ...string) error {
+	res := c.invokeRuntime(ctx, &repb.Command{}, args...)
+	if res.Error != nil {
+		return res.Error
+	}
+	if res.ExitCode != 0 {
+		if len(res.Stderr) > 0 {
+			return fmt.Errorf("%s", strings.TrimSpace(string(res.Stderr)))
+		} else if len(res.Stdout) > 0 {
+			return fmt.Errorf("%s", strings.TrimSpace(string(res.Stdout)))
+		}
+		return fmt.Errorf("exit code %d", res.ExitCode)
+	}
+	return nil
+}
+
+func (c *ociContainer) invokeRuntime(ctx context.Context, command *repb.Command, args ...string) *interfaces.CommandResult {
+	start := time.Now()
+	defer func() {
+		// DO NOT SUBMIT
```

Comment:
?

Reply:
woops - I actually like having these logs while we're developing, so changed to a TODO to remove it later.